lib/SPIRV/PreprocessMetadata.cpp

```diff
@@ -294,12 +294,16 @@ void PreprocessMetadataBase::preprocessOCLMetadata(Module *M, SPIRVMDBuilder *B,
   // !{x} = !{i32 3, i32 102000}
   B->addNamedMD(kSPIRVMD::Source)
       .addOp()
-      .add(CLVer == kOCLVer::CL21 ? spv::SourceLanguageOpenCL_CPP
-                                  : spv::SourceLanguageOpenCL_C)
+      .add(M->getNamedMetadata(kSPIR2MD::OCLCXXVer) &&
+                   (CLVer == kOCLVer::CLCXX10 || CLVer == kOCLVer::CLCXX2021)
+               ? spv::SourceLanguageOpenCL_CPP
```

Comment:
Perhaps we should consider using `SourceLanguageCPP_for_OpenCL` instead, but I'm not entirely sure.  `SourceLanguageOpenCL_CPP` is for the "old" OpenCL C++ that was part of OpenCL 2.2.

Reply:
+@MrSidims, @asudarsa, there are several places using SourceLanguageOpenCL_CPP, should we change to SourceLanguageCPP_for_OpenCL? Thanks very much.

Reply:
I'm also slightly leaning towards SourceLanguageCPP_for_OpenCL, technically it doesn't bring any semantic changes and from debugger perspective CPP_for_OpenCL and OpenCL_CPP are semantically equivalent.

Reply:
We can do this as a separate PR, so not blocking this PR on that.